lib/document/stateTransition/validation/data/validateDocumentsSTDataFactory.js

```diff
@@ -75,6 +78,14 @@ function validateDocumentsSTDataFactory(
               break;
             }
 
+            if (fetchedDocument.getOwnerId() !== document.getOwnerId()
+              || fetchedDocument.getDataContractId() !== document.getDataContractId()
+              || fetchedDocument.entropy !== document.entropy) {
+              result.addError(
+                new DocumentIdComponentsMismatchError(document, fetchedDocument),
```

Comment:
I would have separate errors for the owner and entropy.  We don't need to check data contract id coz it's part of `fetchDocuments` query so if it changed the doc won't be found